dora/core/server/common/src/main/java/alluxio/s3/S3ErrorResponse.java

```diff
@@ -173,4 +183,141 @@ private static Response createErrorResponse(AlluxioRuntimeException e, String re
           s3ErrorCode.getStatus().toString());
     }
   }
+
+  /**
+   * Creates an error response using the given exception.
+   *
+   * @param e a {@link Throwable} object
+   * @param resource an S3 resource key
+   * @return response Http {@link Response}
+   */
+  public static HttpResponse createNettyErrorResponse(Throwable e, String resource) {
+    if (e instanceof AlluxioStatusException) {
+      return createNettyErrorResponse((AlluxioStatusException) e, resource);
+    } else if (e instanceof AlluxioRuntimeException) {
+      return createNettyErrorResponse((AlluxioRuntimeException) e, resource);
+    } else if (e instanceof S3Exception) {
+      return createNettyErrorResponse((S3Exception) e, resource);
+    } else if (e instanceof IOException) {
+      return createNettyErrorResponse((IOException) e, resource);
+    } else {
+      ByteBuf contentBuffer =
```

Comment:
can we put the allocation of `contentBuff` from `Unpooled.copiedBuffer` also into `generateS3ErrorResponse`, so `generateS3ErrorResponse` takes a message String as arg in stead of `ByteBuff`?

ex.

```
else {
  return generateS3ErrorResponse(HttpResponseStatus.INTERNAL_SERVER_ERROR, e.getMessage(),
          HttpHeaderValues.TEXT_PLAIN);
}
```


the reason is netty buffer allocation can be very tricky, so we want to reduce the locations where we allocate Netty ByteBuf (like using `Unpooled.copiedBuffer`).  Typically, Netty ByteBuf requires calling `release` to reclaim the resource, but `Unpooled.copiedBuffer` is an exception. So let's try to consolidate these allocations and it is easier to manage in the future.